api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -79,22 +80,49 @@ default boolean partitionExists(String partitionName) {
   /**
    * Drop a partition with specified name.
    *
-   * @param partitionName The identifier of the partition.
-   * @return true if a partition was deleted, false if the partition did not exist.
+   * @param partitionName the name of the partition
+   * @param ifExists If true, will not throw NoSuchPartitionException if the partition not exists
+   * @return true if a partition was deleted.
+   */
+  boolean dropPartition(String partitionName, boolean ifExists) throws NoSuchPartitionException;
+
+  /**
+   * Drop partitions with specified names.
+   *
+   * @param partitionNames the names of the partition
+   * @param ifExists If true, will not throw NoSuchPartitionException if the partition not exists
+   * @return true if all partitions was deleted.
    */
-  boolean dropPartition(String partitionName);
+  boolean dropPartitions(List<String> partitionNames, boolean ifExists)
+      throws NoSuchPartitionException, UnsupportedOperationException;
 
   /**
    * If the table supports purging, drop a partition with specified name and completely remove
    * partition data by skipping a trash.
    *
    * @param partitionName The name of the partition.
+   * @param ifExists If true, will not throw NoSuchPartitionException if the partition not exists
    * @return true if a partition was deleted, false if the partition did not exist.
    * @throws NoSuchPartitionException If the partition does not exist.
    * @throws UnsupportedOperationException If partition purging is not supported.
    */
-  default boolean purgePartition(String partitionName)
+  default boolean purgePartition(String partitionName, boolean ifExists)
```

Comment:
What happens if you do a purgePartition on a external ta ble? Can you add this to the comments like `purgeTable` did?

Reply:
OK, for external table, purgePartition will only delete metadata in HMS, and not delete HDFS file